E2E managed namespace creation fails if already exists

When running E2E tests locally:

```
> make e2e-local TESTS_MATCH=TestMutationMoreNodes

(...)
2019-08-01T13:55:11.874+0200    DEBUG   e2e     Executing command       {"command": "kubectl create ns e2e-mercury"}
2019-08-01T13:55:12.206+0200    INFO    e2e     Command returned error code     {"command": "kubectl create ns e2e-mercury", "message": "exit status 1"}
Error from server (AlreadyExists): namespaces "e2e-mercury" already exists

2019-08-01T13:55:12.206+0200    ERROR   e2e     Failed to create managed namespace      {"namespace": "e2e-mercury", "error": "command failed: [kubectl create ns e2e-mercury]: ex
it status 1", "errorVerbose": "exit status 1\ncommand failed: [kubectl create ns e2e-mercury]\ngithub.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.(*helper).exec\n\t/Users
/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/run.go:401\ngithub.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.(*helper).kubectl\n\t/Users/seb
/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/run.go:383\ngithub.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.(*helper).createManagedNamespaces\n
\t/Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/run.go:190\ngithub.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.doRun\n\t/Users/seb/wor
k/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/run.go:73\ngithub.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.Command.func2\n\t/Users/seb/work/go/src/
github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/command.go:56\ngithub.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra.(*Command).execute\n\t/Users/seb
/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra/command.go:762\ngithub.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra.(*Com
mand).ExecuteC\n\t/Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra/command.go:852\ngithub.com/elastic/cloud-on-k8s/operators/vendor/
github.com/spf13/cobra.(*Command).Execute\n\t/Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra/command.go:800\nmain.main\n\t/Users/se
b/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/main.go:30\nruntime.main\n\t/usr/local/Cellar/go/1.12.1/libexec/src/runtime/proc.go:200\nruntime.goexit\n\t/u
sr/local/Cellar/go/1.12.1/libexec/src/runtime/asm_amd64.s:1337"}
github.com/elastic/cloud-on-k8s/operators/vendor/github.com/go-logr/zapr.(*zapLogger).Error
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.(*helper).createManagedNamespaces
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/run.go:191
github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.doRun
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/run.go:73
github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run.Command.func2
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run/command.go:56
github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra.(*Command).execute
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra/command.go:762
github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra/command.go:852
github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra.(*Command).Execute
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/spf13/cobra/command.go:800
main.main
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/main.go:30
runtime.main
        /usr/local/Cellar/go/1.12.1/libexec/src/runtime/proc.go:200
2019-08-01T13:55:12.207+0200    INFO    e2e     Cleaning up the scratch directory       {"directory": "/tmp/eck-e2e/e2e"}
Error: command failed: [kubectl create ns e2e-mercury]: exit status 1
```

We probably want one of:

* `kubectl apply`
* get then maybe create
* create but parse `AlreadyExists` error

Which, I think, requires some changes in our `kubectl()` wrapper.

After removing the namespace manually, I now get a `namespace not found` error:

```
⟩ make e2e-local TESTS_MATCH=TestMutationMoreNodes
>>> Running tests in package github.com/elastic/cloud-on-k8s/operators/test/e2e
testing: warning: no tests to run
PASS
ok      github.com/elastic/cloud-on-k8s/operators/test/e2e      0.216s [no tests to run]
>>> Running tests in package github.com/elastic/cloud-on-k8s/operators/test/e2e/apm
testing: warning: no tests to run
PASS
ok      github.com/elastic/cloud-on-k8s/operators/test/e2e/apm  0.212s [no tests to run]
>>> Running tests in package github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd
?       github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd  [no test files]
>>> Running tests in package github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run
?       github.com/elastic/cloud-on-k8s/operators/test/e2e/cmd/run      [no test files]
>>> Running tests in package github.com/elastic/cloud-on-k8s/operators/test/e2e/es
=== RUN   TestMutationMoreNodes
2019-08-01T14:12:16.354+0200    INFO    e2e     Test context initialized        {"context": {"global_operator":{"name":"e2e-global-operator","namespace":"e2e-elastic-system"},"namespace_operators":[{"name":"e2e-mercury-ns-operator","namespace":"e2e-ns-operators","managed_namespace":"e2e-mercury"},{"name":"e2e-venus-ns-operator","namespace":"e2e-ns-operators","managed_namespace":"e2e-venus"}],"e2e_image":"","e2e_namespace":"e2e","e2e_service_account":"e2e","elastic_stack_version":"7.1.1","operator_image":"","test_licence":"","test_regex":"","test_run":"e2e","auto_port_forwarding":true,"local":true}}
=== RUN   TestMutationMoreNodes/K8S_should_be_accessible
=== RUN   TestMutationMoreNodes/Elasticsearch_CRDs_should_exist
=== RUN   TestMutationMoreNodes/Remove_Elasticsearch_if_it_already_exists
Retries (5m0s timeout): .
Retries (5m0s timeout): .
=== RUN   TestMutationMoreNodes/Creating_an_Elasticsearch_cluster_should_succeed
2019-08-01T14:12:18.605+0200    ERROR   stopping early  {"error": "test failure"}
github.com/elastic/cloud-on-k8s/operators/vendor/github.com/go-logr/zapr.(*zapLogger).Error
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/elastic/cloud-on-k8s/operators/test/e2e/test.StepList.RunSequential
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/test/step.go:37
github.com/elastic/cloud-on-k8s/operators/test/e2e/test.RunMutations
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/test/run_mutation.go:36
github.com/elastic/cloud-on-k8s/operators/test/e2e/test.RunMutation
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/test/run_mutation.go:41
github.com/elastic/cloud-on-k8s/operators/test/e2e/es.TestMutationMoreNodes
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/operators/test/e2e/es/mutation_test.go:43
testing.tRunner
        /usr/local/Cellar/go/1.12.1/libexec/src/testing/testing.go:865
--- FAIL: TestMutationMoreNodes (2.25s)
    --- PASS: TestMutationMoreNodes/K8S_should_be_accessible (0.17s)
    --- PASS: TestMutationMoreNodes/Elasticsearch_CRDs_should_exist (0.06s)
    --- PASS: TestMutationMoreNodes/Remove_Elasticsearch_if_it_already_exists (0.20s)
    --- FAIL: TestMutationMoreNodes/Creating_an_Elasticsearch_cluster_should_succeed (0.14s)
        require.go:794:
                Error Trace:    steps_creation.go:25
                Error:          Received unexpected error:
                                namespaces "e2e-mercury" not found
                Test:           TestMutationMoreNodes/Creating_an_Elasticsearch_cluster_should_succeed
FAIL
```

--- a/operators/Makefile
+++ b/operators/Makefile
@@ -305,17 +305,13 @@ e2e-compile:
 # that can reach ES services running in the k8s cluster through port-forwarding.
 LOCAL_E2E_CTX:=/tmp/e2e-local.json
 
-e2e-local: e2e-local-setup
-	@test/e2e/run.sh -run "$(TESTS_MATCH)" -args -testContextPath $(LOCAL_E2E_CTX)
-
-e2e-local-setup: $(LOCAL_E2E_CTX)
-
-$(LOCAL_E2E_CTX):
+e2e-local:
 	@go run test/e2e/cmd/main.go run \
 		--test-run-name=e2e \
 		--test-context-out=$(LOCAL_E2E_CTX) \
 		--auto-port-forwarding \
 		--local
+	@test/e2e/run.sh -run "$(TESTS_MATCH)" -args -testContextPath $(LOCAL_E2E_CTX)
 
 ##########################################
 ##  --    Continuous integration    --  ##

--- a/operators/config/e2e/local_namespace.yaml
+++ b/operators/config/e2e/local_namespace.yaml
@@ -1,0 +1,10 @@
+{{- $testRun := .TestRun -}}
+{{- range .NamespaceOperators }}
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .ManagedNamespace }}
+  labels:
+    test-run: {{ $testRun }}
+{{- end }}

--- a/operators/test/e2e/cmd/run/run.go
+++ b/operators/test/e2e/cmd/run/run.go
@@ -185,16 +185,8 @@ func (h *helper) installCRDs() error {
 }
 
 func (h *helper) createManagedNamespaces() error {
-	for _, ns := range h.managedNamespaces {
-		managedNS := fmt.Sprintf("%s-%s", h.testRunName, ns)
-		if err := h.kubectl("create", "ns", managedNS); err != nil {
-			log.Error(err, "Failed to create managed namespace", "namespace", managedNS)
-			return err
-		}
-		log.Info("Created managed namespace", "namespace", managedNS)
-	}
-
-	return nil
+	log.Info("Creating managed namespaces")
+	return h.kubectlApplyTemplate("config/e2e/local_namespace.yaml", h.testContext, false)
 }
 
 func (h *helper) deployGlobalOperator() error {

